### ISSUE TITLE
Cluster-level mirroring policies

### ISSUE THREAD

**Issue by agrawroh** [envoyproxy/envoy#21131]
## Description

Currently, the mirroring policies are defined on the per-route basis [[Link]](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto) using `request_mirror_policies`. 

We have a use-case where we want to mirror all the traffic from the external [ExtAuthZ](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto#extensions-filters-http-ext-authz-v3-extauthz) and [RateLimit](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ratelimit/v3/rate_limit.proto#envoy-v3-api-msg-extensions-filters-http-ratelimit-v3-ratelimit) services but, there is no good way to achieve it. 

We are working on rewriting these services from scratch (HTTP to gRPC) and want to mirror all the traffic for some time to the old as well as new external services before turning the new services "on" in production.

Is it possible to support mirroring on the cluster-level (all the routes)?

**Comment by lizan** [comment]
Can you elaborate your use case a bit more? IIUC you want the ext_authz and ratelimit traffic to their external server to be mirror-ed?

**Comment by agrawroh** [comment]
> Can you elaborate your use case a bit more? IIUC you want the ext_authz and ratelimit traffic to their external server to be mirror-ed?

@lizan Correct. We are working on rewriting these services from scratch (HTTP to gRPC) and want to mirror all the traffic for some time to the old as well as new service before turning the new services "on" in production.
